packages/core/src/AbstractAppliance.js

```diff
@@ -38,40 +37,69 @@ class AbstractAppliance extends IAppliance {
 		)
 	}
 
-	/**
-	 * Called by a third party to pass data into the appliance.
-	 * If the payload is valid, it is added to the buffer and the appliance is invoked.
-	 *
-	 * @throws
-	 * @param  {[type]} payload [description]
-	 * @return {[type]}         [description]
-	 */
 	ingestPayload = async (payload) => {
-		this.emitter.emit(INPUT_RECEIVED, { payload })
-		assert(isPayloadInstance(payload), 'You cannot ingest data that is not an instance of Payload.')
-		assert(await this.isValidPayload(payload), 'Payload does not satisfy appliance ingestion conditions.')
-		this.payloadBuffer.add(payload)
-		return this.invoke()
+		assert(
+			Payload.isPayload(payload),
+			'You cannot ingest data that is not an instance of Payload.',
+		)
+		assert(
+			await this.isValidPayload(payload),
+			'Payload does not satisfy appliance ingestion conditions.',
+		)
+		this.payloads.insert(payload)
+		const remainingPayloads = await this.invoke(this.payloads)
+		assert(
+			PayloadArray.isPayloadArray(remainingPayloads),
+			'The invoke method was not properly defined; it must return a PayloadArray.',
+		)
+		const werePayloadsProcessed = this.payloads.length !== remainingPayloads.length
+		this.payloads = remainingPayloads
+		return werePayloadsProcessed
 	}
 
-	/**
-	 * Registers a listener to the appliance for a given event type.
-	 *
-	 * Event types are listed in constants/events.js
-	 *
-	 * @param  {String} eventType  The type of event being listened to.
-	 * @param  {Function} listener The listener to be registered for that event.
-	 * @return {EventEmitter}      The EventEmitter (so events can be chained).
-	 */
 	on = (eventType, listener) => this.emitter.on(eventType, listener)
 
-	/**
-	 * Called by an implemented appliance when there is data worth sharing.
-	 *
-	 * @param  {Payload} payload The result that is ready to share.
-	 * @return {Boolean}         Returns true if there are output listeners, false otherwise.
-	 */
-	emitResult = (payload) => this.emitter.emit(OUTPUT, { payload })
+  /**
+   * Emits a `starting` event to listeners.
+   *
+   * @return {Boolean} Returns true if the starting event had listeners, false otherwise.
+   */
+  emitStarting = () => this.emitter.emit(applianceEvents.STARTING)
+
+  /**
+   * Emits a `ready` event to listeners.
+   *
+   * @return {Boolean} Returns true if the ready event had listeners, false otherwise.
+   */
+  emitReady = () => this.emitter.emit(applianceEvents.READY)
```

Comment:
Do we need a separate emitter method for each type of error? This seems repetitive.

Reply:
Good thought.

Reply:
Replaced these with a single `emit` -- a tradeoff is that now implementing appliances are going to need to rely on base-constants, but that's fine.